packages/flutter_tools/test/process_manager_test.dart

```diff
@@ -0,0 +1,208 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:archive/archive.dart';
+import 'package:flutter_tools/src/base/context.dart';
+import 'package:flutter_tools/src/base/logger.dart';
+import 'package:flutter_tools/src/base/os.dart';
+import 'package:flutter_tools/src/base/process.dart';
+import 'package:flutter_tools/src/base/process_manager.dart';
+import 'package:path/path.dart' as path;
+import 'package:test/test.dart';
+
+typedef bool Predicate<T>(T item);
+
+/// Decodes aUTF8-encoded byte array into a list of Strings, where each list
+/// entry represents a line of text.
+List<String> _decode(List<int> data) =>
+    const LineSplitter().convert(UTF8.decode(data));
+
+/// Consumes and returns an entire stream of bytes.
+Future<List<int>> _consume(Stream<List<int>> stream) =>
+    stream.expand((List<int> data) => data).toList();
+
+void main() {
+  group('RecordingProcessManager', () {
+    Directory tmp;
+    ProcessManager manager;
+
+    setUp(() {
+      tmp = Directory.systemTemp.createTempSync('flutter_tools_');
+      manager = new RecordingProcessManager(tmp.path);
+    });
+
+    tearDown(() {
+      tmp.deleteSync(recursive: true);
+    });
+
+    test('start', () async {
+      Process process = await manager.start('which', <String>['bash']);
+      int pid = process.pid;
+      int exitCode = await process.exitCode;
+      List<int> stdout = await _consume(process.stdout);
+      List<int> stderr = await _consume(process.stderr);
+      expect(exitCode, 0);
+      expect(_decode(stdout), <String>['/bin/bash']);
```

Comment:
uber-nit: Incredibly unlikely, but possible to fail depending on `$PATH`. Both MacPorts and Homebrew support installing newer versions of bash, since the one that ships with macOS is ancient (last GPL2 version IIRC). I *suspect* this would fail on my home machine. Maybe `/bin/echo`?